### ISSUE TITLE
[Bug report] bad argument #2 to '?' (number expected, got nil)

### ISSUE THREAD

**Issue by kohane27** [NTBBloodbath/doom-one.nvim#32]
Hello there, hope you're doing well.

I updated today and got the following error:

```
Error detected while processing /home/user/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /home/user/.config/nvim/lua/core/colors.lua:36: Vim(colorscheme):
E5113: Error while calling lua chunk: ...te/pack/packer/start/doom-one.nvim/lua/doom-one/init.lua:
448: bad argument #2 to '?' (number expected, got nil)
stack traceback:
        [builtin#89]: at 0x7f4c2be4a4b0
        ...te/pack/packer/start/doom-one.nvim/lua/doom-one/init.lua:448: in function 'set_colorscheme'
        ...site/pack/packer/start/doom-one.nvim/colors/doom-one.lua:9: in main chunk
        [C]: in function 'cmd'
        /home/user/.config/nvim/lua/core/colors.lua:36: in main chunk
        [C]: in function 'require'
        /home/user/.config/nvim/init.lua:8: in main chunk
stack traceback:
        [C]: in function 'cmd'
        /home/user/.config/nvim/lua/core/colors.lua:36: in main chunk
        [C]: in function 'require'
        /home/user/.config/nvim/init.lua:8: in main chunk
```

The following is the `color.lua` (which I didn't change anything for a long time):

```lua
-- Add color to cursor
vim.g.doom_one_cursor_coloring = false
-- terminal colors
vim.g.doom_one_terminal_colors = false
-- Enable italic comments
vim.g.doom_one_italic_comments = false
-- Enable TS support
vim.g.doom_one_enable_treesitter = true
-- Color whole diagnostic text or only underline
vim.g.doom_one_diagnostics_text_color = false
-- Enable transparent background
vim.g.doom_one_transparent_background = false

-- Pumblend transparency
vim.g.doom_one_pumblend_enable = false
vim.g.doom_one_pumblend_transparency = 20

-- Plugins integration
vim.g.doom_one_plugin_nvim_tree = true
vim.g.doom_one_plugin_whichkey = true
vim.g.doom_one_plugin_indent_blankline = true
vim.g.doom_one_plugin_lspsaga = true
vim.g.doom_one_plugin_barbar = true
vim.g.doom_one_plugin_neorg = false
vim.g.doom_one_plugin_telescope = false
vim.g.doom_one_plugin_neogit = false
vim.g.doom_one_plugin_dashboard = false
vim.g.doom_one_plugin_startify = false
vim.g.doom_one_plugin_vim_illuminate = false

vim.cmd([[colorscheme doom-one]])
```

```
:version
NVIM v0.8.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser
```

Any help is much appreciated. Thank you!

**Comment by luisvgs** [comment]
I also got this error after the update.

**Comment by NTBBloodbath** [comment]
Hey, hope you're doing well too!

This is very odd to me, line 448 shouldn't be sending an error to you as you're disabling neorg integration and that code chunk shouldn't be running in your Neovim instances.

If you're compiling Neovim from source can you please build using `RelWithDebInfo` instead of just `Release`? That build mode will be the same as `Release` but will include debugging information that is useful to debug Lua errors traceback so Lua isn't going to send that `'?'` nonsense thanks to a unknown debugging symbol in the error message as this isn't useful for us to get what's wrong.

```
bad argument #2 to '?' (number expected, got nil)
```

Cheers!

**Comment by phush0** [comment]
Rebuild this morning from master and had same problem. But disabling neorg fixed it for me. Like something had changed in nightly builds.

**Comment by luisvgs** [comment]
@phush0 this approach does not works for me :(

**Comment by jdhao** [comment]
Same error here on macOS with nvim stable release (0.8.0)

To reproduce, `nvim -u init.vim`, content of `init.vim`:

```
set runtimepath+=/Users/jdhao/.local/share/nvim/site/pack/packer/opt/doom-one.nvim
colorscheme doom-one
```

**Comment by kohane27** [comment]
```
:version
NVIM v0.9.0-dev-314+ge8cc489ac
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wn
o-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-form
at-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=alwa
ys -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/
home/usr/bin/neovim/build/cmake.config -I/home/usr/bin/neovim/src -I/home/usr/bin/neovim/.deps/usr/include -I/usr/include -I
/home/usr/bin/neovim/build/src/nvim/auto -I/home/usr/bin/neovim/build/include
Compiled by usr@usr

Features: +acl +iconv +tui
```

```
Error detected while processing /home/usr/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /home/usr/.config/nvim/lua/core/colors.lua:36: Vim(colorscheme):E5113: Error while calling lua chunk: ...te/pack/packer/start/doom-one.nvim/lua/doom-one/init.lua:448: bad argument #2 to '?' (number expected, got nil)
stack traceback:
        [builtin#89]: at 0x5584a634b480
        ...te/pack/packer/start/doom-one.nvim/lua/doom-one/init.lua:448: in function 'set_colorscheme'
        ...site/pack/packer/start/doom-one.nvim/colors/doom-one.lua:9: in main chunk
        [C]: in function 'cmd'
        /home/usr/.config/nvim/lua/core/colors.lua:36: in main chunk
        [C]: in function 'require'
        /home/usr/.config/nvim/init.lua:8: in main chunk
stack traceback:
        [C]: in function 'cmd'
        /home/usr/.config/nvim/lua/core/colors.lua:36: in main chunk
        [C]: in function 'require'
        /home/usr/.config/nvim/init.lua:8: in main chunk
```

Hopefully it helps.

**Comment by fitrh** [comment]
> line 448 shouldn't be sending an error to you as you're disabling neorg integration

@NTBBloodbath every `vim.g.doom_one_*` should be set to `false` by
default (or a more reliable mechanism instead of using `or`) since doing
`false or true` will always give us `true`

**Comment by NTBBloodbath** [comment]
Fixed on #33
